pyro/infer/abstract_infer.py

```diff
@@ -151,4 +151,4 @@ def __call__(self, *args, **kwargs):
         if not isinstance(logits, torch.autograd.Variable):
             logits = Variable(logits)
         ix = dist.Categorical(logits=logits).sample()
-        return traces[ix.data[0]]
+        return traces[ix]
```

Comment:
Why not `ix.item()` here? (This may not be perfectly tested)

Reply:
Because a scalar is a valid index value. We can change it `ix.item()` but it's not necessary.

Reply:
This is great! It will be so much easier to write mixture models now.

Reply:
Completely; scalars will simplify a number of clunky looking stuff.